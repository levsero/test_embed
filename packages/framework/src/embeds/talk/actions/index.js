import {
  MICROPHONE_MUTED,
  MICROPHONE_UNMUTED,
  RECORDING_CONSENT_ACCEPTED,
  RECORDING_CONSENT_DENIED,
  CALL_ENDED,
  CALL_FAILED,
  CALL_STARTED,
  RESET_CALL_FAILED,
  INCREMENT_CALL_TIMER,
} from './action-types'

let callInterval

const startCallCounter = () => (dispatch) => {
  if (!callInterval) {
    callInterval = setInterval(() => {
      dispatch({ type: INCREMENT_CALL_TIMER })
    }, 1000)
  }
}

const stopCallCounter = () => {
  clearInterval(callInterval)
  callInterval = null
}

export const muteMicrophone = () => ({
  type: MICROPHONE_MUTED,
})

export const unmuteMicrophone = () => ({
  type: MICROPHONE_UNMUTED,
})

export const acceptRecordingConsent = () => ({
  type: RECORDING_CONSENT_ACCEPTED,
})
export const declineRecordingConsent = () => ({
  type: RECORDING_CONSENT_DENIED,
})

export const callStarted = () => (dispatch) => {
  dispatch({ type: CALL_STARTED })
  dispatch(startCallCounter())
}

export const callEnded = () => (dispatch) => {
  dispatch({ type: CALL_ENDED })
  stopCallCounter()
}

export const callFailed = () => (dispatch) => {
  dispatch({ type: CALL_FAILED })
  stopCallCounter()
}

export const resetCallFailed = () => ({
  type: RESET_CALL_FAILED,
})
