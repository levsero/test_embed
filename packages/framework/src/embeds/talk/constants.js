export const CONTACT_OPTIONS = {
  CALLBACK_ONLY: 'widget/talk/CALLBACK_ONLY',
  PHONE_ONLY: 'widget/talk/PHONE_ONLY',
  CALLBACK_AND_PHONE: 'widget/talk/CALLBACK_AND_PHONE',
  CLICK_TO_CALL: 'widget/talk/CLICK_TO_CALL'
}

export const CAPABILTY_NAMES = {
  'widget/talk/CALLBACK_ONLY': 'Request a callback',
  'widget/talk/PHONE_ONLY': 'Call us',
  'widget/talk/CALLBACK_AND_PHONE': 'Request a callback and call us',
  'widget/talk/CLICK_TO_CALL': 'Click to call'
}

export const TEST_IDS = {
  BUTTON_HANG_UP: 'button-hang-up',
  ICON_AVATAR: 'Icon--avatar'
}
