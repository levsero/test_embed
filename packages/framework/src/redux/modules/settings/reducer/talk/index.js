import _ from 'lodash'
import { UPDATE_SETTINGS } from 'src/redux/modules/settings/settings-action-types'

const initialState = {
  title: {},
  suppress: false,
  nickname: null,
}

const talkSettings = (state = initialState, action) => {
  const { type, payload } = action

  switch (type) {
    case UPDATE_SETTINGS:
      return {
        nickname: _.get(payload, 'webWidget.talk.nickname', state.nickname),
        suppress: _.get(payload, 'webWidget.talk.suppress', state.suppress),
        title: _.get(payload, 'webWidget.talk.title', state.title),
      }
    default:
      return state
  }
}

export default talkSettings
