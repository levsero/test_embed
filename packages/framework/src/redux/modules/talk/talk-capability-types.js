export const CALLBACK_ONLY = 'widget/talk/CALLBACK_ONLY'
export const PHONE_ONLY = 'widget/talk/PHONE_ONLY'
export const CALLBACK_AND_PHONE = 'widget/talk/CALLBACK_AND_PHONE'
export const CLICK_TO_CALL = 'widget/talk/CLICK_TO_CALL'

export const CAPABILITY_TYPE_CODES = {
  CALLBACK_ONLY: 0,
  PHONE_ONLY: 1,
  CALLBACK_AND_PHONE: 2,
  CLICK_TO_CALL: 3,
}
