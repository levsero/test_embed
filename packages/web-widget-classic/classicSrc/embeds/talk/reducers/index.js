import embeddedVoiceCallStatus from 'classicSrc/embeds/talk/reducers/embedded-voice-call-status'
import microphoneMuted from 'classicSrc/embeds/talk/reducers/microphone-muted'
import recordingConsent from 'classicSrc/embeds/talk/reducers/recording-consent'
import agentAvailability from 'classicSrc/embeds/talk/reducers/talk-agent-availability'
import averageWaitTime from 'classicSrc/embeds/talk/reducers/talk-average-wait-time'
import callback from 'classicSrc/embeds/talk/reducers/talk-callback'
import embeddableConfig from 'classicSrc/embeds/talk/reducers/talk-embeddable-config'
import formState from 'classicSrc/embeds/talk/reducers/talk-form-state'
import isPolling from 'classicSrc/embeds/talk/reducers/talk-is-polling'
import vendor from 'classicSrc/embeds/talk/reducers/talk-vendor'
import timeInCall from 'classicSrc/embeds/talk/reducers/time-in-call'
import { combineReducers } from 'redux'

export default combineReducers({
  agentAvailability,
  averageWaitTime,
  callback,
  embeddableConfig,
  embeddedVoiceCallStatus,
  formState,
  isPolling,
  microphoneMuted,
  recordingConsent,
  timeInCall,
  vendor,
})
