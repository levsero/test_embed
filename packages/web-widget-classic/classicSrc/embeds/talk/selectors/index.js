import { i18n } from 'classicSrc/app/webWidget/services/i18n'
import { CONTACT_OPTIONS } from 'classicSrc/embeds/talk/constants'
import {
  CALLBACK_ONLY,
  CALLBACK_AND_PHONE,
  CLICK_TO_CALL,
} from 'classicSrc/embeds/talk/talk-capability-types'
import { getSettingsTalkTitle } from 'classicSrc/redux/modules/settings/settings-selectors'
import _ from 'lodash'
import { createSelector } from 'reselect'
import isFeatureEnabled from '@zendesk/widget-shared-services/feature-flags'

export const getEmbeddedVoiceSupported = (_state) =>
  isFeatureEnabled(null, 'embedded_voice_enabled')

export const getIsCallInProgress = (state) => state.talk.embeddedVoiceCallStatus.isCallInProgress
export const getHasLastCallFailed = (state) => state.talk.embeddedVoiceCallStatus.hasLastCallFailed
export const getRecordingConsent = (state) => state.talk.recordingConsent
export const getUserRecordingConsentRequirement = (state) =>
  state.talk.embeddableConfig.recordingConsent
export const getMicrophoneMuted = (state) => state.talk.microphoneMuted
export const getTimeInCall = (state) => state.talk.timeInCall
export const getTalkEmbeddableConfig = (state) => state.talk.embeddableConfig
export const getPhoneNumber = createSelector(
  getTalkEmbeddableConfig,
  (config) => config.phoneNumber
)
export const getIsEmbeddedVoiceEnabled = (state) =>
  getTalkEmbeddableConfig(state)?.capability === CLICK_TO_CALL
export const getEmbeddableConfigEnabled = (state) => getTalkEmbeddableConfig(state).enabled
export const getDeferredStatusOnline = (state) =>
  getTalkEmbeddableConfig(state).deferredStatusOnline
export const getTalkEmbeddableConfigConnected = (state) => getTalkEmbeddableConfig(state).connected
export const getAgentAvailability = (state) => state.talk.agentAvailability
export const getFormState = (state) => state.talk.formState
export const getCallback = (state) => state.talk.callback
export const getAverageWaitTime = (state) => state.talk.averageWaitTime.waitTime
export const getAverageWaitTimeEnabled = (state) => state.talk.averageWaitTime.enabled
export const getSocketIoVendor = (state) => state.talk.vendor.io
export const getIsPollingTalk = (state) => state.talk.isPolling

export const isCallbackEnabled = (state) => {
  const { capability } = state.talk.embeddableConfig

  return _.includes([CALLBACK_ONLY, CALLBACK_AND_PHONE], capability)
}

export const getAverageWaitTimeString = createSelector(
  [getAverageWaitTimeEnabled, getAverageWaitTime],
  (averageWaitTimeEnabled, averageWaitTime) => {
    const averageWaitTimeNumber = parseInt(averageWaitTime, 10)

    if (!averageWaitTimeEnabled || averageWaitTimeNumber === 0) return null

    const waitTimeForm = averageWaitTimeNumber > 1 ? 'Plural' : 'Singular'

    return i18n.t(`embeddable_framework.talk.form.averageWaitTime${waitTimeForm}`, {
      averageWaitTime,
    })
  }
)

export const getCapability = createSelector(
  [getTalkEmbeddableConfig, getEmbeddedVoiceSupported],
  (talkConfig, embeddedVoiceSupported) => {
    return embeddedVoiceSupported ? CLICK_TO_CALL : talkConfig.capability
  }
)

export const getOfflineTitle = (state) => {
  const capability = getCapability(state)
  const title = i18n.getSettingTranslation(getSettingsTalkTitle(state))

  if (title) {
    return title
  }

  switch (capability) {
    case CONTACT_OPTIONS.PHONE_ONLY:
      return i18n.t('embeddable_framework.talk.phoneOnly.title')
    case CONTACT_OPTIONS.CLICK_TO_CALL:
      return i18n.t('embeddable_framework.talk.embeddedVoice.channel.title')
    case CONTACT_OPTIONS.CALLBACK_AND_PHONE:
    case CONTACT_OPTIONS.CALLBACK_ONLY:
    default:
      return i18n.t('embeddable_framework.talk.form.title')
  }
}

export const getTalkTitleKey = createSelector(
  [getCapability, isCallbackEnabled],
  (capability, callbackEnabled) => {
    if (capability === CLICK_TO_CALL) return 'embeddable_framework.talk.embeddedVoice.channel.title'
    if (callbackEnabled) {
      return 'embeddable_framework.launcher.label.talk.request_callback'
    } else {
      return 'embeddable_framework.launcher.label.talk.call_us'
    }
  }
)
