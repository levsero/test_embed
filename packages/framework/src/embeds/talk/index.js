import PropTypes from 'prop-types'
import { Component, lazy } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { Route, Switch, Redirect } from 'react-router-dom'
import LoadingPage from 'components/LoadingPage'
import { WidgetThemeProvider } from 'src/components/Widget'
import SuspensePage from 'src/components/Widget/SuspensePage'
import { getCapability, getIsCallInProgress } from 'src/embeds/talk/selectors'
import { loadTalkVendors } from 'src/redux/modules/talk'
import {
  getAgentAvailability,
  getSocketIoVendor,
  getDeferredStatusOnline,
} from 'src/redux/modules/talk/talk-selectors'
import { CONTACT_OPTIONS } from './constants'
import SuccessNotificationPage from './pages/SuccessNotificationPage'
import OfflinePage from './pages/offline/OfflinePage'
import CallbackAndPhonePage from './pages/online/CallbackAndPhonePage'
import CallbackOnlyPage from './pages/online/CallbackOnlyPage'
import PhoneOnlyPage from './pages/online/PhoneOnlyPage'
import routes from './routes'

const EmbeddedVoicePage = lazy(() =>
  import(/* webpackChunkName: 'lazy/talk/click_to_call' */ './pages/online/EmbeddedVoicePage')
)

const onlineContactOptions = {
  [CONTACT_OPTIONS.CALLBACK_ONLY]: routes.callbackOnly(),
  [CONTACT_OPTIONS.PHONE_ONLY]: routes.phoneOnly(),
  [CONTACT_OPTIONS.CALLBACK_AND_PHONE]: routes.callbackAndPhone(),
  [CONTACT_OPTIONS.CLICK_TO_CALL]: routes.clickToCall(),
}

// This component needs to be a class component since the parent WebWidget component expects to be able
// to put a ref on each embed component.
class Talk extends Component {
  componentDidMount() {
    if (!this.props.talkVendorLoaded) {
      this.props.loadTalkVendors()
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.talkIsDeferred || prevProps.talkIsDeferred) return
    if (
      prevProps.contactOption != this.props.contactOption ||
      prevProps.hasAvailableAgents != this.props.hasAvailableAgents
    ) {
      this.resetRoutes()
    }
  }

  resetRoutes() {
    this.props.history.replace(routes.home())
  }

  render() {
    const {
      hasAvailableAgents,
      contactOption,
      talkIsDeferred,
      isEmbeddedVoiceCallInProgress,
    } = this.props

    if (talkIsDeferred) {
      return <LoadingPage />
    }

    return (
      <WidgetThemeProvider>
        <SuspensePage>
          <Switch>
            <Route path={routes.successNotification()} component={SuccessNotificationPage} />

            <Route path={routes.online()}>
              <Switch>
                <Route path={routes.callbackOnly()} component={CallbackOnlyPage} />
                <Route path={routes.phoneOnly()} component={PhoneOnlyPage} />
                <Route path={routes.callbackAndPhone()} component={CallbackAndPhonePage} />
                <Route path={routes.clickToCall()} component={EmbeddedVoicePage} />

                <Redirect to={onlineContactOptions[contactOption]} />
              </Switch>
            </Route>

            <Route path={routes.offline()}>
              <Route component={OfflinePage} />
            </Route>

            <Redirect
              to={
                hasAvailableAgents || isEmbeddedVoiceCallInProgress
                  ? routes.online()
                  : routes.offline()
              }
            />
          </Switch>
        </SuspensePage>
      </WidgetThemeProvider>
    )
  }
}

Talk.propTypes = {
  hasAvailableAgents: PropTypes.bool.isRequired,
  isEmbeddedVoiceCallInProgress: PropTypes.bool.isRequired,
  contactOption: PropTypes.oneOf(Object.values(CONTACT_OPTIONS)).isRequired,
  history: PropTypes.shape({
    replace: PropTypes.func.isRequired,
  }),
  loadTalkVendors: PropTypes.func.isRequired,
  talkVendorLoaded: PropTypes.bool.isRequired,
  talkIsDeferred: PropTypes.bool.isRequired,
}

const mapStateToProps = (state) => ({
  hasAvailableAgents: getAgentAvailability(state),
  isEmbeddedVoiceCallInProgress: getIsCallInProgress(state),
  contactOption: getCapability(state),
  talkVendorLoaded: !!getSocketIoVendor(state),
  talkIsDeferred: getDeferredStatusOnline(state),
})

const actionCreators = {
  loadTalkVendors,
}

const connectedComponent = connect(mapStateToProps, actionCreators, null, { forwardRef: true })(
  withRouter(Talk)
)

export { connectedComponent as default, Talk as Component }
