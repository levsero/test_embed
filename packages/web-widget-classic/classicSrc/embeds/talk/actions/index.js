import { TALK_SUCCESS_DONE_BUTTON_CLICKED } from 'classicSrc/embeds/talk/action-types'
import {
  TALK_EMBEDDABLE_CONFIG_SOCKET_EVENT,
  TALK_AGENT_AVAILABILITY_SOCKET_EVENT,
  TALK_AVERAGE_WAIT_TIME_SOCKET_EVENT,
  TALK_DISCONNECT_SOCKET_EVENT,
  UPDATE_CALLBACK_FORM,
  TALK_CALLBACK_REQUEST,
  TALK_CALLBACK_SUCCESS,
  TALK_CALLBACK_FAILURE,
  TALK_VENDOR_LOADED,
  RECEIVED_DEFERRED_TALK_STATUS,
} from 'classicSrc/embeds/talk/action-types'
import {
  BASE_TALK_POLL_INTERVAL,
  MAX_TALK_POLL_INTERVAL,
  REQUESTS_BEFORE_BACKOFF,
} from 'classicSrc/embeds/talk/constants'
import { getFormState, getIsPollingTalk } from 'classicSrc/embeds/talk/selectors'
import { handleCloseButtonClicked, updateBackButtonVisibility } from 'classicSrc/redux/modules/base'
import {
  getTalkEnabled,
  getTalkNickname,
  getTalkServiceUrl,
  getDeferredTalkApiUrl,
} from 'classicSrc/redux/modules/selectors'
import { http, socketio } from 'classicSrc/service/transport'
import wait from 'classicSrc/util/wait'
import _ from 'lodash'
import { parseUrl, errorTracker } from '@zendesk/widget-shared-services'
import {
  MICROPHONE_MUTED,
  MICROPHONE_UNMUTED,
  RECORDING_CONSENT_ACCEPTED,
  RECORDING_CONSENT_DENIED,
  CALL_ENDED,
  CALL_FAILED,
  CALL_STARTED,
  RESET_CALL_FAILED,
  INCREMENT_CALL_TIMER,
} from './action-types'

let callInterval

const startCallCounter = () => (dispatch) => {
  if (!callInterval) {
    callInterval = setInterval(() => {
      dispatch({ type: INCREMENT_CALL_TIMER })
    }, 1000)
  }
}

const stopCallCounter = () => {
  clearInterval(callInterval)
  callInterval = null
}

export const muteMicrophone = () => ({
  type: MICROPHONE_MUTED,
})

export const unmuteMicrophone = () => ({
  type: MICROPHONE_UNMUTED,
})

export const acceptRecordingConsent = () => ({
  type: RECORDING_CONSENT_ACCEPTED,
})
export const declineRecordingConsent = () => ({
  type: RECORDING_CONSENT_DENIED,
})

export const callStarted = () => (dispatch) => {
  dispatch({ type: CALL_STARTED })
  dispatch(startCallCounter())
}

export const callEnded = () => (dispatch) => {
  dispatch({ type: CALL_ENDED })
  stopCallCounter()
}

export const callFailed = () => (dispatch) => {
  dispatch({ type: CALL_FAILED })
  stopCallCounter()
}

export const resetCallFailed = () => ({
  type: RESET_CALL_FAILED,
})

export function updateTalkEmbeddableConfig(config) {
  return {
    type: TALK_EMBEDDABLE_CONFIG_SOCKET_EVENT,
    payload: config,
  }
}

export function updateTalkAgentAvailability(availability) {
  return {
    type: TALK_AGENT_AVAILABILITY_SOCKET_EVENT,
    payload: availability,
  }
}

export function updateTalkAverageWaitTime(averageWaitTime) {
  return {
    type: TALK_AVERAGE_WAIT_TIME_SOCKET_EVENT,
    payload: averageWaitTime,
  }
}

export function talkDisconnect() {
  return {
    type: TALK_DISCONNECT_SOCKET_EVENT,
  }
}

export function updateTalkCallbackForm(formState) {
  return {
    type: UPDATE_CALLBACK_FORM,
    payload: formState,
  }
}

export function submitTalkCallbackForm(serviceUrl, nickname) {
  return (dispatch, getState) => {
    const formState = getFormState(getState())
    const additionalInfo = _.pickBy(
      {
        name: formState.name,
        description: formState.description,
      },
      _.identity
    )
    const subdomain = parseUrl(serviceUrl).hostname.split('.')[0]
    const params = {
      phoneNumber: formState.phone,
      additionalInfo,
      subdomain,
      keyword: nickname,
    }
    const callbacks = {
      done: () => {
        dispatch({ type: TALK_CALLBACK_SUCCESS, payload: formState })
        dispatch(updateTalkCallbackForm({}))
        dispatch(updateBackButtonVisibility(false))
      },
      fail: (err) => {
        const errorMessage = err.responseJSON
          ? err.responseJSON.error
          : JSON.parse(err.response.text).error
        const error = {
          message: errorMessage,
          status: err.status,
        }

        dispatch({
          type: TALK_CALLBACK_FAILURE,
          payload: error,
        })
      },
    }

    dispatch({
      type: TALK_CALLBACK_REQUEST,
      payload: formState,
    })

    http.callMeRequest(serviceUrl, {
      params,
      callbacks,
    })
  }
}

let requests = 0
const talkPollInterval = () => {
  if (requests < REQUESTS_BEFORE_BACKOFF) {
    requests += 1
    return BASE_TALK_POLL_INTERVAL
  }
  const delay = BASE_TALK_POLL_INTERVAL * Math.pow(2, requests - REQUESTS_BEFORE_BACKOFF)
  requests += 1
  return Math.min(delay, MAX_TALK_POLL_INTERVAL)
}

const pathsToSkip = {}
export function pollTalkStatus() {
  return async (dispatch, getState) => {
    const path = getDeferredTalkApiUrl(getState())
    while (getIsPollingTalk(getState())) {
      const skip = (document.hidden && requests > 1) || pathsToSkip[path]

      if (!skip) {
        http
          .get({ path }, { skipCache: true })
          .then((response) => {
            if (!getIsPollingTalk(getState())) {
              return
            }
            dispatch({
              type: RECEIVED_DEFERRED_TALK_STATUS,
              payload: response.body,
            })
          })
          .catch((err) => {
            errorTracker.warn(err, {
              rollbarFingerprint: 'Failed to connect to deferred talk endpoint',
              rollbarTitle: 'Failed to connect to deferred talk endpoint',
            })
            if (err.status == 404) {
              pathsToSkip[path] = true
            }
          })
      }
      await wait(talkPollInterval())
    }
  }
}

export function loadTalkVendors() {
  return (dispatch, getState) => {
    if (!getTalkEnabled(getState())) return

    const nickname = getTalkNickname(getState())

    const onSuccess = ({ default: io }) => {
      dispatch(handleTalkVendorLoaded({ io }))
      if (_.isEmpty(nickname)) return

      const socket = socketio.connect(io, getTalkServiceUrl(getState()), nickname)

      socketio.mapEventsToActions(socket, { dispatch })
    }
    const onFailure = (err) => {
      errorTracker.error(err)
    }

    return import(/* webpackChunkName: 'talk-sdk' */ 'socket.io-client')
      .then(onSuccess)
      .catch(onFailure)
  }
}

export function handleTalkVendorLoaded(vendor) {
  return {
    type: TALK_VENDOR_LOADED,
    payload: vendor,
  }
}

export const successDoneButtonClicked = () => (dispatch) => {
  dispatch(handleCloseButtonClicked())
  dispatch({
    type: TALK_SUCCESS_DONE_BUTTON_CLICKED,
  })
}
