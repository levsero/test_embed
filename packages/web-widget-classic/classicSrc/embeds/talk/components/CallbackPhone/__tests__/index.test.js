import { render } from '@testing-library/react'
import { CALLBACK_AND_PHONE } from 'classicSrc/embeds/talk/talk-capability-types'
import { Component as CallbackPhone } from '../'

jest.mock('classicSrc/embeds/talk/components/CallbackForm', () => () => (
  <div data-testid="talk--callbackForm" />
))

test('renders the component', () => {
  const props = {
    phoneNumber: '1234567890',
    formattedPhoneNumber: '+1 2345 67890',
    phoneLabel: 'Our phone number',
    capability: CALLBACK_AND_PHONE,
  }
  const { container } = render(<CallbackPhone {...props} />)

  expect(container).toMatchSnapshot()
})
