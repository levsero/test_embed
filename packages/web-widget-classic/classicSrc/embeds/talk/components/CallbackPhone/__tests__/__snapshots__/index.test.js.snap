// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
.c1 {
  color: #1f73b7;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover,
.c1:focus,
.c1:active {
  color: #1f73b7;
}

.c0 {
  margin-bottom: 20px;
}

<div>
  <div
    class="c0"
  >
    Our phone number 
    <a
      class="c1"
      data-testid="talk-phone-number"
      dir="ltr"
      href="tel:1234567890"
      target="_blank"
    >
      +1 2345 67890
    </a>
  </div>
</div>
`;
